backend/src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -42,12 +44,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       json.addProperty("loggedIn", true);
       json.addProperty("userEmail", userEmail);
       json.addProperty("logoutUrl", logoutUrl);
+
+      Entity userEntity = UserCrud.readEntity("email", userEmail, "User");
+      if (userEntity == null) {
+        userEntity = UserCrud.createUser(userEmail);
+      }
```

Comment:
nit: Since you're not doing anything with userEntity...

```suggestion
      if (UserCrud.readEntity("email", userEmail, "User") == null) {
       UserCrud.createUser(userEmail);
      }
```